summoner-cli/src/Summoner/Project.hs

```diff
@@ -155,25 +157,61 @@ generateProject settingsNoUpload isOffline projectName Config{..} = do
+                        successMessage ("The default nix package set " <> showNixPkgSet defaultNixPkgSet <> " will be used in the project")
+                        pure $ Just defaultNixPkgSet
+                chooseYesNo (mkDefaultYesNoPrompt "nix package set") yesDo noDo
+            Last set@(Just n) -> set <$ successMessage ("The nix package set " <> showNixPkgSet n <> " will be used in the project")
+        else pure Nothing
+
     -- get what build tool to use in the project
     -- If user chose only one during CLI, we assume to use only that one.
-    getCabalStack :: (Decision, Decision) -> IO (Bool, Bool)
-    getCabalStack = \case
-        (Idk, Idk) -> decisionToBool cCabal (mkDefaultYesNoPrompt "cabal") >>= \c ->
-            if c then decisionToBool cStack (mkDefaultYesNoPrompt "stack") >>= \s -> pure (c, s)
-            else stackMsg True >> pure (False, True)
-        (Nop, Nop) -> errorMessage "Neither cabal nor stack was chosen" >> exitFailure
-        (Yes, Yes) -> output (True, True)
-        (Yes, _)   -> output (True, False)
-        (_, Yes)   -> output (False, True)
-        (Nop, Idk) -> output (False, True)
-        (Idk, Nop) -> output (True, False)
+    getCabalStackNix :: (Decision, Decision, Decision) -> IO (Bool, Bool, Bool)
+    getCabalStackNix = \case
+        (Idk, Idk, Idk) -> do
+            c <- decisionToBool cCabal (mkDefaultYesNoPrompt "cabal")
+            s <- decisionToBool cStack (mkDefaultYesNoPrompt "stack")
+            n <- decisionToBool cNix (mkDefaultYesNoPrompt "nix")
+            when (not c && n) nixWithoutCabalError
+            output (c, s, n)
+        (Nop, Nop, Nop) -> noneOfError
+        (Yes, Yes, Yes) -> output (True, True, True)
+        (Idk, Idk, Nop) -> output (True, False, False)
+        (Idk, Idk, Yes) -> output (True, False, True)
+        (Idk, Nop, _)   -> output (True, False, False)
+        (Idk, Yes, _)   -> output (True, True, False)
+        (Nop, Idk, _)   -> output (False, True, False)
+        (Nop, Nop, Idk) -> noneOfError
+        (Nop, Nop, Yes) -> nixWithoutCabalError
+        (Nop, Yes, _)   -> output (False, True, False)
+        (Yes, Idk, _)   -> output (True, False, False)
+        (Yes, Nop, _)   -> output (True, False, False)
+        (Yes, Yes, Idk) -> output (True, True, False)
+        (Yes, Yes, Nop) -> output (True, True, False)
       where
-        output :: (Bool, Bool) -> IO (Bool, Bool)
-        output x@(c, s) = cabalMsg c >> stackMsg s >> pure x
+        output :: (Bool, Bool, Bool) -> IO (Bool, Bool, Bool)
+        output x@(c, s, n) = cabalMsg c >> stackMsg s >> nixMsg n >> pure x
 
         cabalMsg c = targetMessageWithText c "Cabal" "used in this project"
         stackMsg c = targetMessageWithText c "Stack" "used in this project"
+        nixMsg   c = targetMessageWithText c "Nix"   "used in this project"
+
+        nixWithoutCabalError = errorMessage "Using nix without cabal is an unsupported configuration" >> exitFailure
```

Comment:
Why this is the issue?
The `cabal` file should be created in any case. This mostly will affect other configs like Travis etc.

Reply:
> Thanks for taking your time to answer all these questions! I appreciate this!
> 
> > The generated nix configuration calls cabal2nix ('translates' .cabal to a .nix) automatically for the user, so it needs a cabal file for the project, otherwise, it can't function.
> 
> Stack also needs `cabal` file, and we still can choose one: if you choose only Stack you still have `cabal` file, but also `stack.yaml` files, and Travis config will be only Stack related (without Cabal runs). So the same logic can be applied to Nix & Cabal

Ah, okay, that will probably simplify the 27-case function then.